instrumentation/redisson/redisson-common/testing/src/main/java/io/opentelemetry/javaagent/instrumentation/redisson/AbstractRedissonAsyncClientTest.java

```diff
@@ -0,0 +1,253 @@
+import org.redisson.config.Config;
+import org.redisson.config.SingleServerConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.GenericContainer;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public abstract class AbstractRedissonAsyncClientTest {
+
+  private static final Logger logger =
+      LoggerFactory.getLogger(AbstractRedissonAsyncClientTest.class);
+
+  @RegisterExtension
+  protected static final InstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  private static final GenericContainer<?> redisServer =
+      new GenericContainer<>("redis:6.2.3-alpine").withExposedPorts(6379);
+
+  private static int port;
+
+  private static String address;
+  private static RedissonClient redisson;
+
+  @BeforeAll
+  static void setupAll() {
+    redisServer.start();
+    port = redisServer.getMappedPort(6379);
+    address = "localhost:" + port;
+  }
+
+  @AfterAll
+  static void cleanupAll() {
+    redisson.shutdown();
+    redisServer.stop();
+  }
+
+  @BeforeEach
+  void setup() throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+    String newAddress = address;
+    if (useRedisProtocol()) {
+      // Newer versions of redisson require scheme, older versions forbid it
+      newAddress = "redis://" + address;
+    }
+    Config config = new Config();
+    SingleServerConfig singleServerConfig = config.useSingleServer();
+    singleServerConfig.setAddress(newAddress);
+    singleServerConfig.setTimeout(30_000);
+    try {
+      // disable connection ping if it exists
+      singleServerConfig
+          .getClass()
+          .getMethod("setPingConnectionInterval", int.class)
+          .invoke(singleServerConfig, 0);
+    } catch (NoSuchMethodException e) {
+      logger.warn("no setPingConnectionInterval method", e);
+    }
+    redisson = Redisson.create(config);
+    testing.clearData();
+  }
+
+  @Test
+  void futureSet() throws ExecutionException, InterruptedException, TimeoutException {
+    RBucket<String> keyObject = redisson.getBucket("foo");
+    RFuture<Void> future = keyObject.setAsync("bar");
+    future.get(30, TimeUnit.SECONDS);
+
+    testing.waitAndAssertSortedTraces(
+        orderByRootSpanKind(SpanKind.INTERNAL, SpanKind.CLIENT),
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("SET")
+                        .hasKind(CLIENT)
+                        .hasAttribute(SemanticAttributes.NET_SOCK_PEER_ADDR, "127.0.0.1")
+                        .hasAttribute(SemanticAttributes.NET_SOCK_PEER_NAME, "localhost")
+                        .hasAttribute(SemanticAttributes.NET_SOCK_PEER_PORT, (long) port)
+                        .hasAttribute(SemanticAttributes.DB_SYSTEM, "redis")
+                        .hasAttribute(SemanticAttributes.DB_STATEMENT, "SET foo ?")
+                        .hasAttribute(SemanticAttributes.DB_OPERATION, "SET")));
```

Comment:
You can use `hasAttributesSatisfyingExactly()` to assert the exact set of attributes observed:
```suggestion
                        .hasAttributesSatisfyingExactly(
                          equalTo(SemanticAttributes.NET_SOCK_PEER_ADDR, "127.0.0.1"),
                          equalTo(SemanticAttributes.NET_SOCK_PEER_NAME, "localhost"),
                          equalTo(SemanticAttributes.NET_SOCK_PEER_PORT, (long) port),
                          equalTo(SemanticAttributes.DB_SYSTEM, "redis"),
                          equalTo(SemanticAttributes.DB_STATEMENT, "SET foo ?"),
                          equalTo(SemanticAttributes.DB_OPERATION, "SET")));
```

Reply:
Done